pkg/config/config.go

```diff
@@ -195,6 +195,11 @@ func WatchControllerConfig(ctr controller.Controller, mgr manager.Manager) error
 		return err
 	}
 
+	err = fillRegistryIfNecessary(nonCachedClient, configMap)
```

Comment:
I'm wondering how this works now that plugin registry is optional.

Reply:
it only works if a given env variable is passed to the controller deployment with the name of te registry K8S service.
This allows setting this env variable in the deployment section of the CSV for OLM.

Reply:
But if we rely on the environment variable and do the check every time, why not just store the registry URL internally rather than updating a configmap. It's strange to have a deployment managing its own configmap in this way, since data should generally flow from a configmap to the deployment, not vice-versa.

Reply:
It's just building the default registry URL from the service name when the config map doesn't exist (=> default value), which will be the case with OLM deployments.